syntax = "proto3";

package evaluation.common;

option go_package = "github.com/genx/evaluation/grpc/common";

// Common message types used across all metric services

message EvaluationItem {
  string question = 1;
  string answer = 2;
  string context = 3;
  string expected_answer = 4;
  string reference_output = 5;
  map<string, string> metadata = 6;
}

message MetricResponse {
  double score = 1;
  bool success = 2;
  string error_message = 3;
  map<string, string> metadata = 4;
  string metric_name = 5;
  double execution_time_ms = 6;
}

message BatchItem {
  string item_id = 1;
  EvaluationItem evaluation_data = 2;
}

message BatchItemResult {
  string item_id = 1;
  string question = 2;
  map<string, double> metric_scores = 3;
  bool success = 4;
  string error_message = 5;
}

message HealthCheckRequest {
  string service = 1;
}

message HealthCheckResponse {
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
  }
  ServingStatus status = 1;
  string message = 2;
}
